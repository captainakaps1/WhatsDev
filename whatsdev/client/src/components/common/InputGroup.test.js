import InputGroup from "./InputGroup";
import { findByTestId, render } from "@testing-library/react";

describe("Component Test - InputGroup", () => {
  const data = {
    name: "",
    placeholder: "",
    value: "",
    error: "",
    onChange: () => {},
    icon: "",
    type: "",
  };
  it("matches snapshot", () => {
    const { asFragment } = render(<InputGroup {...data} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without crashing", async () => {
    const { findByTestId } = render(<InputGroup {...data} />);
    expect(await findByTestId("input-group-div")).toBeDefined();
  });
});
