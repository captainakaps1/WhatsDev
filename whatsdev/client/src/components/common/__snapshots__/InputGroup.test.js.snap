// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Test - InputGroup matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="input-group mb-3"
    data-testid="input-group-div"
  >
    <div
      class="input-group-prepend"
    >
      <span
        class="input-group-text"
      >
        <i
          class=""
        />
      </span>
    </div>
    <input
      class="form-control form-control-lg"
      name=""
      placeholder=""
      type=""
      value=""
    />
  </div>
</DocumentFragment>
`;
